src/H5T.c

```diff
@@ -5606,7 +5606,7 @@ H5T_nameof(const H5T_t *dt)
             HGOTO_ERROR(H5E_DATATYPE, H5E_CANTINIT, NULL, "not a named datatype")
         case H5T_STATE_NAMED:
         case H5T_STATE_OPEN:
-            ret_value = &(dt->path);
+            ret_value = (H5G_name_t *)&(dt->path); /* TODO: should not cast away const! */
```

Comment:
Please don't cast away const, it's hiding the underlying design issue, which should be fixed.